### PR TITLE
feat(util/gvalid): supporting set field bail and use json tag as alias name for validation

### PR DESCRIPTION
**Pull request by whtkim**
**gvalid.New().FieldBail():** supporting set field bail for all validation rules
**gvalid.New().JsonTagAsAlias():** supporting use json tag as alias name for validation when not set alias name in validate rules and json tag is exists

### CODE REVIEW

**Comment by hailaz** [comment]
请提交相关测试示例，以保证代码完整可用并具有使用的参考性。

File: util/gvalid/gvalid_validator.go [hunk 1]
```diff
@@ -28,8 +28,10 @@ type Validator struct {
 	ruleFuncMap                       map[string]RuleFunc // ruleFuncMap stores custom rule functions for current Validator.
 	useAssocInsteadOfObjectAttributes bool                // Using `assoc` as its validation source instead of attribute values from `Object`.
 	bail                              bool                // Stop validation after the first validation error.
+	fieldBail                         bool                // Stop field validation after the first validation error.
```
Comment (gqcn) [inline util/gvalid/gvalid_validator.go:31]: The attribute `fieldBail` might be duplicated with attribute `bail`.

File: util/gvalid/gvalid_validator.go [hunk 2]
```diff
@@ -124,6 +133,13 @@ func (v *Validator) Ci() *Validator {
 	return newValidator
 }
 
+// JsonTagAsAlias sets using the JSON tag value as an alias name.
```
Comment (gqcn) [inline util/gvalid/gvalid_validator.go:136]: FieldBail  sets using the JSON tag value as alias when the alias name is not set in the validation rule.

File: util/gvalid/gvalid_validator_check_struct.go [hunk 3]
```diff
@@ -132,6 +132,12 @@ func (v *Validator) doCheckStruct(ctx context.Context, object interface{}) Error
 			fieldName       = field.Name()                  // Attribute name.
 			name, rule, msg = ParseTagValue(field.TagValue) // The `name` is different from `attribute alias`, which is used for validation only.
 		)
+		// It uses json tag value as alias name if exists.
+		if len(name) == 0 && v.jsonTagAsAlias {
+			if jsonTag := field.Tag("json"); len(jsonTag) > 0 {
```
Comment (gqcn) [inline util/gvalid/gvalid_validator_check_struct.go:137]: - Please consider there's json tag value pattern like `json:"name,omitempty"`
- Please add associated unit testing case for your changes.

**Review by gqcn — CHANGES_REQUESTED** [review]

File: util/gvalid/gvalid_validator.go (hunk 1 above)
Comment (whtkim) [reply util/gvalid/gvalid_validator.go:31]: The attribute `fieldBail` is equivalent to adding bail to each field of the struct

File: util/gvalid/gvalid_validator.go (hunk 2 above)
Comment (whtkim) [reply util/gvalid/gvalid_validator.go:136]: fixed

**Comment by hailaz** [comment]
ci fail https://github.com/gogf/gf/actions/runs/10146655300/job/28055228639?pr=3715
```
Running [/home/runner/golangci-lint-1.58.2-linux-amd64/golangci-lint run  --timeout 3m0s] in [/home/runner/work/gf/gf] ...
  Received [27](https://github.com/gogf/gf/actions/runs/10146655300/job/28055228639?pr=3715#step:4:29)6880 of 276880 (100.0%), 0.3 MBs/sec
  util/gvalid/gvalid_validator_check_struct.go:22: Function 'doCheckStruct' is too long (336 > 330) (funlen)
  func (v *Validator) doCheckStruct(ctx context.Context, object interface{}) Error {
  
  level=warning msg="[config_reader] The configuration option `linters.errcheck.ignore` is deprecated, please use `linters.errcheck.exclude-functions`."
``

**Comment by gqcn** [comment]
Closed due to inactivity for too long.

**Pull request closed without merging**